TradeItIosAdSdk/AdService.swift

```diff
@@ -125,6 +126,10 @@ class AdService {
         return UIScreen.main.bounds.width
     }
 
+    static func deviceId() -> String {
+        return UIDevice.current.identifierForVendor?.uuidString ?? ""
```

Comment:
Interestingly the `deviceId` is unique per app vendor so we cannot do a frequency cap using this across partners.